src/Export/Adapters/ExportItemAdapter.php

```diff
@@ -142,16 +147,27 @@ public function adaptProduct(Item $item, ProductEntity $product): ?Item
     public function adaptVariant(Item $item, ProductEntity $product): ?Item
     {
         if ($this->eventDispatcher) {
-            $this->eventDispatcher->dispatch(new BeforeVariantAdaptEvent($product, $item), BeforeVariantAdaptEvent::NAME);
+            $this->eventDispatcher->dispatch(
+                new BeforeVariantAdaptEvent($product, $item),
+                BeforeVariantAdaptEvent::NAME,
+            );
         }
 
         try {
             foreach ($this->adapterFactory->getOrderNumbersAdapter()->adapt($product) as $orderNumber) {
                 $item->addOrdernumber($orderNumber);
             }
 
-            foreach ($this->adapterFactory->getAttributeAdapter()->adapt($product) as $attribute) {
-                $item->addMergedAttribute($attribute);
+            if ($this->pluginConfig->getMainVariant() != 'default') {
+                foreach ($this->adapterFactory->getOptionsAdapter()->getOptionAttributes($product) as $attribute) {
+                    $item->addMergedAttribute($attribute);
+                }
+            } else {
+                $attributes = $this->adapterFactory->getVariantConfigurationAdapter()
+                    ->getOptionAttributes($product);
+                foreach ($attributes as $attribute) {
+                    $item->addMergedAttribute($attribute);
+                }
```

Comment:
The initial idea was to just call the `optionsAdapter` within the `adaptProduct` function and the `variantConfigurationAdapter` within the `adaptVariant` function.
There should be no config check needed within this class.